android/src/main/java/com/reactlibrary/JudoReactNativeModule.java

```diff
@@ -2,526 +2,447 @@
+            case "PLN":
+                return Currency.PLN;
+            case "SAR":
+                return Currency.SAR;
+            case "SEK":
+                return Currency.SEK;
+            case "SGD":
+                return Currency.SGD;
+            case "QAR":
+                return Currency.QAR;
+            case "USD":
+                return Currency.USD;
+            case "ZAR":
+                return Currency.ZAR;
+            default:
+                return Currency.GBP;
         }
-        WritableMap result = new WritableNativeMap();
-        result.putString("postCodeCheck", risks.getPostCodeCheck());
-        return result;
     }
 
-    private WritableMap convert(OrderDetails orderDetails) {
-        if (orderDetails == null) {
-            return null;
-        }
-        WritableMap result = new WritableNativeMap();
-        result.putString("orderId", orderDetails.getOrderId());
-        result.putString("orderStatus", orderDetails.getOrderStatus().name());
-        result.putString("orderFailureReason", orderDetails.getOrderFailureReason());
-        result.putString("timestamp", orderDetails.getTimestamp());
-        return result;
+    private Reference getReference(ReadableMap options) {
+        ReadableMap configurations = options.getMap("configuration");
+        ReadableMap reference = configurations.getMap("reference");
+
+        String consumerReference = reference.getString("consumerReference");
+        String paymentReference = reference.getString("paymentReference");
+
+        // TODO: MAP METADATA BUNDLE ?
+        return new Reference.Builder()
+                .setConsumerReference(consumerReference)
+                .setPaymentReference(paymentReference)
+                .build();
     }
 
-    private Judo getJudo(ReadableMap options) {
-        Bundle bundle = new Bundle();
-        ReadableMap metadataMap = options.getMap("metaData");
-        if (metadataMap != null) {
-            for (String keyName : metadataMap.toHashMap().keySet()) {
-                bundle.putString(keyName, metadataMap.getString(keyName));
-            }
+    private String getSiteId(ReadableMap options) {
+        ReadableMap configurations = options.getMap("configuration");
+        return configurations.getString("siteId");
+    }
+
+    private CardNetwork[] getCardNetworks(ReadableMap options) {
+
+        int CARD_VISA = 1;
+        int CARD_MASTERCARD = 1 << 1;
+        int CARD_MAESTRO = 1 << 2;
+        int CARD_AMEX = 1 << 3;
+        int CARD_CHINAUNIONPAY = 1 << 4;
+        int CARD_JCB = 1 << 5;
+        int CARD_DISCOVER = 1 << 6;
+        int CARD_DINERSCLUB = 1 << 7;
+        int CARD_ALL = 1 << 8;
+
+        ReadableMap configurations = options.getMap("configuration");
+        int cardNetworkValue = configurations.getInt("supportedCardNetworks");
+
+        ArrayList<CardNetwork> supportedNetworks = new ArrayList<>();
+
+        if ((cardNetworkValue & CARD_ALL) == 1 << 8) {
```

Comment:
why not to: `(cardNetworkValue & CARD_ALL) == CARD_ALL` ?

Reply:
Changed. Resolve if fixed issue.